File: 그리디(Greedy)/체육복.py
def solution(n, lost, reserve):
    answer = 0
    data=[]
    count=0
    for i in range (31):
        data.append(1)
    for i in reserve:
        data[i-1]+=1
    for i in lost:
        data[i-1]-=1


    for i in range (n):
        if data[i]==2:
            if data[i-1]==0:
                data[i-1]+=1
                data[i]-=1
            elif data[i+1]==0:
                data[i+1]+=1
                data[i]-=1


    for i in range (n):
        if data[i]==0:
            count+=1

    return n-count

print(solution(5,[1,2,5],[2,4]))

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int[] students=new int[n+2];
        int answer=n;
        
        Arrays.fill(students,1);
        students[0]=0;
        students[n+1]=0;
        
        for (int l:lost){
            students[l]-=1;
        }
        
        for (int r:reserve){
            students[r]+=1;
        }
        
        for (int i=1;i<=n;i++){
            if (students[i]==0){
                if (students[i-1]==2){
                    students[i-1]-=1;
                    students[i]=1;
                }else if(students[i+1]==2){
                    students[i+1]-=1;
                    students[i]=1;
                }
            }
            if (students[i]==0){
                answer-=1;
            }
        }
        
        
        return answer;
    }
}